### PR TITLE
CORCI-727: Fix "Skip-{build,test}: true" handling

### PR DESCRIPTION
**Pull request by brianjmurrell**
Using Skip-build: true and Skip-test: true in commit messages doesn't
result in all build and test stages being skipped.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -873,6 +944,14 @@ pipeline {
             }
         }
         stage('Unit Test') {
+            when {
+                beforeAgent true
+                // expression { skipTest != true }
+                expression {
+                    sh script: 'git show -s --format=%B | grep "^Skip-test: true"',
```
Comment (JohnMalmberg) [inline Jenkinsfile]: Staging needs to skip tests also.
                    if [ "${NO_CI_TESTING}" == 'true' ]; then ...

File: Jenkinsfile [hunk 2]
```diff
@@ -965,6 +1044,14 @@ pipeline {
             }
         }
         stage('Test') {
+            when {
+                beforeAgent true
+                // expression { skipTest != true }
+                expression {
+                    sh script: 'git show -s --format=%B | grep "^Skip-test: true"',
```
Comment (JohnMalmberg) [inline Jenkinsfile]: ditto

File: Jenkinsfile (hunk 1 above)
Comment (brianjmurrell) [reply Jenkinsfile]: But isn't skipping tests on staging already working?

File: Jenkinsfile (hunk 2 above)
Comment (brianjmurrell) [reply Jenkinsfile]: Ditto.

File: Jenkinsfile (hunk 1 above)
Comment (JohnMalmberg) [reply Jenkinsfile]: Skipping tests on staging is skipping implemented in the runTest script.

So if this change only affects tests run in containers, than nothing special for changing is needed.

File: Jenkinsfile (hunk 2 above)
Comment (JohnMalmberg) [reply Jenkinsfile]: Same as above.

File: Jenkinsfile (hunk 1 above)
Comment (brianjmurrell) [reply Jenkinsfile]: I don't think whether what's being skipped is in a container is relevant.  I think the overall point is that these changes shouldn't be changing anything in the way staging's skipping is working.  I think the [run results on staging](https://build-stage.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-879/3/pipeline) confirm that.

**Comment by brianjmurrell** [comment]
RPM test failures are due to mercury update in progress.

File: src/client/dfs/dfuse_hl.c [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /**
- * (C) Copyright 2018 Intel Corporation.
+ * (C) Copyright 2018Copyright 2018-2019 Intel Corporation.
```
Comment (JohnMalmberg) [inline src/client/dfs/dfuse_hl.c]: Looks like an editing issue here.

File: src/common/drpc.c [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * (C) Copyright 2018 Intel Corporation.
+ * (C) Copyright 2018Copyright 2018-2019 Intel Corporation.
```
Comment (JohnMalmberg) [inline src/common/drpc.c]: Same.

File: src/client/dfs/dfuse_hl.c (hunk 3 above)
Comment (brianjmurrell) [reply src/client/dfs/dfuse_hl.c]: Yeah.  Files are getting edited without having copyrights updated and so when I merge, my git hook tries to update them.  It shouldn't on merges, and I have remedied that, but people need to be updating those.

We ought to add the check for that in our checkpatch/linting step.

File: src/common/drpc.c (hunk 4 above)
Comment (brianjmurrell) [reply src/common/drpc.c]: Ditto.

**Pull request merged**